cpplint.py

```diff
@@ -4761,23 +4750,18 @@ def CheckBraces(filename, clean_lines, linenum, error):
     # No control clauses with braces should have its contents on the same line
     # Exclude } which will be covered by empty-block detect
     # Exclude ; which may be used by while in a do-while
-    if keyword := re.search(
-        r"\b(else if|if|while|for|switch)"  # These have parens
-        r"\s*\(.*\)\s*(?:\[\[(?:un)?likely\]\]\s*)?{\s*[^\s\\};]",
-        line,
-    ):
-        error(
-            filename,
-            linenum,
-            "whitespace/newline",
-            5,
-            f"Controlled statements inside brackets of {keyword.group(1)} clause"
-            " should be on a separate line",
+    if (
+        keyword := re.search(
+            r"\b(else if|if|while|for|switch)"  # These have parens
+            r"\s*\(.*\)\s*(?:\[\[(?:un)?likely\]\]\s*)?{\s*[^\s\\};]",
+            line,
+        )
+    ) or (
+        keyword := re.search(
+            r"\b(else|do|try)"  # These don't have parens
+            r"\s*(?:\[\[(?:un)?likely\]\]\s*)?{\s*[^\s\\}]",
+            line,
```

Comment:
I wonder why it put everything in brackets...

Reply:
Assignment Operator may require parentheses.  Also there is a RUF rule that adds parens to clarify precedence between and, or, not.